definitions/features/upstream.rb

```diff
@@ -9,6 +9,58 @@ class Features::Upstream < ForemanMaintain::Feature
   end
 
   def setup_repositories(_version)
-    raise NotImplementedError
+    activation_key = ENV['EXTERNAL_KATELLO_ACTIVATION_KEY']
+    org = ENV['EXTERNAL_KATELLO_ORG']
+    if activation_key
+      use_activation_key(activation_key, org)
+    else
+      update_foreman_release_pkg
+      update_katello_release_pkg
+    end
+  end
+
+  def update_foreman_release_pkg
+    pkg_url = feature(:upstream_repositories).foreman_release_pkg_url(version)
+    if el?
+      update_release_pkg_el(pkg_url)
+    elsif debian_or_ubuntu?
+      update_release_pkg_deb(pkg_url)
+    end
+  end
+
+  def update_katello_release_pkg
+    if feature(:katello)
+      pkg_url = feature(:upstream_repositories).katello_release_pkg(version)
+      update_release_pkg_el(pkg_url)
+    end
+  end
+
+  def use_activation_key(activation_key, org)
+    org_options = org ? %(--org #{shellescape(org)}) : ''
+    execute!(%(subscription-manager register #{org_options}\
+                --activationkey #{shellescape(activation_key)} --force))
+  end
+
+  def update_release_pkg_el(pkg_url)
+    package_manager.install(pkg_url, assumeyes: true)
+  end
+
+  def update_release_pkg_deb(pkg_url)
+    # apt don't support pkg installation using URL
+    exit_status, = execute_with_status("wget -q -P /tmp/ #{pkg_url}")
+    if exit_status == 0
+      package_manager.install('/tmp/foreman-release.deb', assumeyes: true)
+    else
+      warn! "Couldn't install Foreman Release Package: #{pkg_url}"
+    end
+  end
+
+  def el_repositories(el_major_version)
+    if rhel?
```

Comment:
I thought to add the repository validation, but feels like we can get cases where repository labels are not standard.. so stopped here, maybe we should not do that and let the yum/dnf/apt fail if it does not find correct repositories?